client/service/src/task_manager/tests.rs

```diff
@@ -315,92 +319,92 @@ fn ensure_task_manager_future_continues_when_childs_not_essential_task_fails() {
-	let (subscriber, spans_entered) = setup_subscriber();
+fn subprocess_telemetry_span_is_forwarded_to_task() {
+	if env::var("SUBPROCESS_TEST").is_err() {
+		return;
+	}
+
+	let (subscriber, spans_found) = setup_subscriber();
+	tracing_log::LogTracer::init().unwrap();
 	let _sub_guard = tracing::subscriber::set_global_default(subscriber);
 
-	let telemetry_span = TelemetrySpan::new();
+	let mut runtime = tokio::runtime::Runtime::new().unwrap();
 
-	let span = tracing::info_span!("test");
-	let _enter = span.enter();
+	let prefix_span = tracing::info_span!("prefix");
+	let _enter_prefix_span = prefix_span.enter();
+
+	let telemetry_span = TelemetrySpan::new();
+	let _enter_telemetry_span = telemetry_span.enter();
 
-	let mut runtime = tokio::runtime::Runtime::new().unwrap();
 	let handle = runtime.handle().clone();
 	let task_executor = TaskExecutor::from(move |fut, _| handle.spawn(fut).map(|_| ()));
-	let task_manager = TaskManager::new(task_executor, None, Some(telemetry_span.clone())).unwrap();
+	let task_manager = new_task_manager(task_executor);
 
 	let (sender, receiver) = futures::channel::oneshot::channel();
-	let spawn_handle = task_manager.spawn_handle();
 
-	let span = span.clone();
 	task_manager.spawn_handle().spawn(
-		"test",
+		"log-something",
 		async move {
-			assert_eq!(span, Span::current());
-			spawn_handle.spawn("test-nested", async move {
-				assert_eq!(span, Span::current());
-				sender.send(()).unwrap();
-			}.boxed());
-		}.boxed(),
+			log::info!("boo!");
+			sender.send(()).unwrap();
+		}
+		.boxed(),
 	);
 
-	// We need to leave exit the span here. If tokio is not running with multithreading, this
-	// would lead to duplicate spans being "active" and forwarding the wrong one.
-	drop(_enter);
 	runtime.block_on(receiver).unwrap();
 	runtime.block_on(task_manager.clean_shutdown());
-	drop(runtime);
-
-	let spans = spans_entered.lock();
-	// We entered the telemetry span and the "test" in the future, the nested future and
-	// the "test" span outside of the future. So, we should have recorded 3 spans.
-	assert_eq!(5, spans.len());
-
-	assert_eq!(spans[0], "test");
-	assert_eq!(spans[1], telemetry_span.span().metadata().unwrap().name());
-	assert_eq!(spans[2], "test");
-	assert_eq!(spans[3], telemetry_span.span().metadata().unwrap().name());
-	assert_eq!(spans[4], "test");
+
+	let spans = spans_found.lock().take().unwrap();
+	assert_eq!(2, spans.len());
+
+	assert_eq!(spans[0], prefix_span.id().unwrap());
+	assert_eq!(spans[1], telemetry_span.span().id().unwrap());
+}
+
+#[test]
+fn telemetry_span_is_forwarded_to_task() {
+	let executable = env::current_exe().unwrap();
+	let output = std::process::Command::new(executable)
+		.env("SUBPROCESS_TEST", "1")
+		.args(&["--nocapture", "subprocess_telemetry_span_is_forwarded_to_task"])
+		.output()
+		.unwrap();
+	println!("{}", String::from_utf8(output.stdout).unwrap());
+	eprintln!("{}", String::from_utf8(output.stderr).unwrap());
```

Comment:
See comment above: not sure what the point of printing is if we don't assert on it.

Reply:
it's because if the test fail it will show up :brain: clever right?? :grin: